### PR TITLE
Fixed Console warnings for Terra-list & Terra-button 

### PR DESCRIPTION
**Pull request by gt106551**
### Summary
Resolved console warnings related to Terra-list and Terra-button.

**What was changed:**
Removed `ariaDescription` prop from list
Removed `isSelectable` & `isSelected` custom prop from Button

**Why it was changed:**

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

[UXPLATFORM-9893]()
---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-list/src/List.jsx [hunk 1]
```diff
@@ -34,18 +34,6 @@ const propTypes = {
...
-   * References a visible or visually hidden text element
-   * on the page that conveys information about the list
-   * and how to interact with it. Maps to
-   * [aria-details](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-details).
-   * Can be used alongside `ariaDescribedBy` or `ariaDescription`.
-   * `aria-details` is not supported in all screen reader/browser
-   * combinations. Consumers should test for different screen
-   * reader/browser combinations to ensure that instructions
-   * are consistently read.
-   */
```
Comment (supreethmr) [inline packages/terra-list/src/List.jsx:48]: you need to remove comments added for ariaDescription not the comments added for ariaDetails

Comment (gt106551) [reply packages/terra-list/src/List.jsx:48]: Updated.

**Comment by supreethmr** [comment]
I can still there is warning for isFocusDisabled prop on list subcomponents 
Screenshot 2023-11-30 at 11 57 24 AM

**Comment by gt106551** [comment]
> I can still there is warning for isFocusDisabled prop on list subcomponents Screenshot 2023-11-30 at 11 57 24 AM

Fixed.

File: packages/terra-list/src/ListSection.jsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-param-reassign */
```
[the remark this answers is not in the document]
Comment (gt106551) [reply packages/terra-list/src/ListSection.jsx]: Updated.

File: packages/terra-core-docs/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Removed
+  * Removed `ariaDescription` prop from list
```
Comment (supreethmr) [inline packages/terra-core-docs/CHANGELOG.md:6]:
```suggestion
* Changed
  * Removed `ariaDescription` example added for terra-list
```

File: packages/terra-list/src/ListSectionHeader.jsx [hunk 4]
```diff
@@ -110,7 +110,8 @@ const ListSectionHeader = ({
     attrSpread.onBlur = ListUtils.wrappedEventCallback(onBlur, event => event.currentTarget.setAttribute('data-item-show-focus', 'true'));
     attrSpread.onMouseDown = ListUtils.wrappedEventCallback(onMouseDown, event => event.currentTarget.setAttribute('data-item-show-focus', 'false'));
   }
-
+  /* eslint-disable-next-line */
```
Comment (supreethmr) [inline packages/terra-list/src/ListSectionHeader.jsx:113]: what are you disabling here

File: packages/terra-list/src/ListSubsection.jsx [hunk 5]
```diff
@@ -81,7 +81,8 @@ const ListSubsection = ({
 }) => {
   const [listItemNodes, setlistItemNodes] = useState(children);
   let listSubSectionItemNode = useRef();
-
+  /* eslint-disable-next-line */
```
Comment (supreethmr) [inline packages/terra-list/src/ListSubsection.jsx:84]:
```suggestion
// eslint-disable-next-line no-param-reassign
```

File: packages/terra-list/src/ListSectionHeader.jsx (hunk 4 above)
Comment (gt106551) [reply packages/terra-list/src/ListSectionHeader.jsx]: Due to no-param-reassign error i have added this.

File: packages/terra-list/src/ListSubsection.jsx (hunk 5 above)
Comment (gt106551) [reply packages/terra-list/src/ListSubsection.jsx]: updated

**Review by supreethmr — APPROVED** [review]

File: packages/terra-list/src/List.jsx (hunk 1 above)
Comment (sugan2416) [reply packages/terra-list/src/List.jsx:48]: Remove description of `ariaDescription` and restore description of `ariaDetails`. 
`ariaDetails` prop needs to be uncommented.

**Comment by sugan2416** [comment]
This example can be removed since `aria-description` is no longer used: packages/terra-core-docs/src/terra-dev-site/test/list/AriaDescriptionList.test.jsx
Check for usages in spec and remove its tests if present.

File: packages/terra-list/src/List.jsx (hunk 1 above)
Comment (gt106551) [reply packages/terra-list/src/List.jsx:48]: Updated..

**Comment by gt106551** [comment]
> This example can be removed since `aria-description` is no longer used: packages/terra-core-docs/src/terra-dev-site/test/list/AriaDescriptionList.test.jsx Check for usages in spec and remove its tests if present.

Updated.

**Review by sugan2416 — APPROVED** [review]

**Pull request merged**